spec/lib/castle/extractors/headers_spec.rb

```diff
@@ -1,8 +1,9 @@
 # frozen_string_literal: true
 
 describe Castle::Extractors::Headers do
-  subject(:headers) { described_class.new(request).call }
+  subject(:headers) { described_class.new(formatted_headers).call }
 
+  let(:formatted_headers) { Castle::HeaderFilter.new(request).call }
```

Comment:
I'd prefer to remove `env` and `request` from this spec and just pass properly formatted headers instead